File: ios/MySwiftThingy.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MySwiftThingy, NSObject)

RCT_EXTERN_METHOD(callbackMethod:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(simpleMethod:(NSString *)message)

@end

File: ios/MySwiftThingy.swift
import Foundation

@objc(MySwiftThingy)
class MySwiftThingy: NSObject {

  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc func callbackMethod(callback: RCTResponseSenderBlock) -> Void {
    let resultsDict = [
      "success" : true
    ];

    callback([NSNull() ,resultsDict])
  }

  @objc func simpleMethod(message: String!) {
    print("\(message)")
  }
}
